### PR TITLE
gui

### PR DESCRIPTION
**Pull request by MrFast-js**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -76,6 +90,7 @@ val remapJar by tasks.named<net.fabricmc.loom.task.RemapJarTask>("remapJar") {
     archiveClassifier.set("all")
     from(tasks.shadowJar)
     input.set(tasks.shadowJar.get().archiveFile)
+    destinationDir = file("C:/Users/jedij/AppData/Roaming/.minecraft/mods")
```
Comment (Flou21) [inline build.gradle.kts:93]: that will not work, at least i think so

File: src/main/java/de/torui/coflsky/Config.java [hunk 2]
```diff
@@ -3,4 +3,6 @@
 public class Config {
     public static final int KeepFlipsForSeconds = 42;
     public static String BaseUrl = "https://sky.coflnet.com";
+
+
```
Comment (Ekwav) [inline src/main/java/de/torui/coflsky/Config.java:7]:
```suggestion
```

File: src/main/java/de/torui/coflsky/handlers/EventRegistry.java [hunk 3]
```diff
@@ -236,7 +245,6 @@ public void onGuiOpen(GuiOpenEvent event) {
 	@SubscribeEvent(priority = EventPriority.HIGHEST)
 	public void onItemTooltipEvent(ItemTooltipEvent event) {
 		if (!config.extendedtooltips) return;
-		if(descriptionHandler == null) return;
```
Comment (Ekwav) [inline src/main/java/de/torui/coflsky/handlers/EventRegistry.java:239]: why was this removed?

File: src/main/java/de/torui/coflsky/gui/CoflGui.java [hunk 4]
```diff
@@ -0,0 +1,471 @@
...
+                        value=value.toString().replaceAll("[^0-9m.btk]","");
+                        if(!value.toString().equals(setting.get("value").toString()) && value.toString().length()>0) {
+                            System.out.println("DIFFERENCE '"+value+"' '"+setting.get("value")+"'");
+                            updateSetting(setting.get("key").getAsString(),value);
+                        }
+                        return Unit.INSTANCE;
+                    });
+                }
+                index++;
+            }
```
Comment (Ekwav) [inline src/main/java/de/torui/coflsky/gui/CoflGui.java:445]: could you split this into multiple functions to fit on one screen each?

**Review by Ekwav — CHANGES_REQUESTED** [review]

File: finish.bat [hunk 5]
```diff
@@ -0,0 +1,6 @@
+powershell -window hidden -Command
```
[the remark this answers is not in the document]
Comment (MrFast-js) [reply finish.bat:1]: you can delete it it was to restart my game when the mod compiled

File: src/main/java/de/torui/coflsky/CoflSkyCommand.java [hunk 6]
```diff
@@ -138,10 +140,9 @@ public void processCommand(ICommandSender sender, String[] args) throws CommandE
 					SendCommandToServer(args, sender);
 					return;
 				}
-			} 
-			
-			else {
-				ListHelp(sender);
+			} else {
+				GuiUtil.open(new CoflGui(true));
+				// ListHelp(sender);
```
[the remark this answers is not in the document]
Comment (MrFast-js) [reply src/main/java/de/torui/coflsky/CoflSkyCommand.java:145]: that could be it or you could make the help menu ListHelp(sender) activate when do '/cofl help'

File: src/main/java/de/torui/coflsky/handlers/EventRegistry.java (hunk 3 above)
Comment (MrFast-js) [reply src/main/java/de/torui/coflsky/handlers/EventRegistry.java:239]: I didn't purposefully remove it may have been an accident or I made the fork before that line existed

File: src/main/java/de/torui/coflsky/gui/CoflGui.java (hunk 4 above)
Comment (MrFast-js) [reply src/main/java/de/torui/coflsky/gui/CoflGui.java:445]: sure

**Review by Ekwav — CHANGES_REQUESTED** [review]
Can't start the mod, crashing with:
```
---- Minecraft Crash Report ----
// On the bright side, I bought you a teddy bear!

Time: 17.02.23 17:24
Description: There was a severe problem during mod loading that has caused the game to fail

net.minecraftforge.fml.common.LoaderException: java.lang.NoClassDefFoundError: de/torui/coflsky/gui/CoflGui
	at net.minecraftforge.fml.common.LoadController.transition(LoadController.java:162)
	at net.minecraftforge.fml.common.Loader.initializeMods(Loader.java:739)
	at net.minecraftforge.fml.client.FMLClientHandler.finishMinecraftLoading(FMLClientHandler.java:310)
	at net.minecraft.client.Minecraft.func_71384_a(Minecraft.java:495)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:329)
	at net.minecraft.client.main.Main.main(SourceFile:124)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)
Caused by: java.lang.NoClassDefFoundError: de/torui/coflsky/gui/CoflGui
	at de.torui.coflsky.CoflSky.init(CoflSky.java:100)
```

File: build.gradle.kts [hunk 7]
```diff
@@ -47,13 +52,22 @@ dependencies {
     mappings("de.oceanlabs.mcp:mcp_stable:22-1.8.9")
     forge("net.minecraftforge:forge:1.8.9-11.15.1.2318-1.8.9")
 
+    // Add essential dependency so we can access elementa (Gui creator)
```
Comment (Ekwav) [inline build.gradle.kts:55]: We avoided dependencies like this so far, can this be done without including all of essential?

**Review by Ekwav — CHANGES_REQUESTED** [review]

**Comment by Ekwav** [comment]
I am still not accepting adding essentials as a dependency
